Add VSCode settings

emackey opened the pull request:
This adds a `.vscode` settings folder, and a setup guide for VSCode.

Also, this adds a missing npm task `startPublic` for launching `node server.js --public`.

pjcozzi commented on Documentation/Contributors/VSCodeGuide/README.md:
```diff
@@ -0,0 +1,55 @@
+# VSCode Guide
```
Add a link to this from https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors#contributor-guides

emackey replied:
Added, thanks.

pjcozzi commented:
Perhaps add this to CHANGES.md.

pjcozzi commented:
Comments are not part of JSON, I suppose they were autogenerated and work OK with VSCode?

pjcozzi commented:
Can someone with VSCode test and merge?

emackey commented:
Comments are apparently valid in VSCode config files, and yes they are auto-generated.

emackey commented:
Forum post:  https://groups.google.com/forum/#!topic/cesium-dev/aUD6QGxuLrQ

shunter commented:
I tried this out this afternoon.  I hadn't used VS Code before, but it seems to be a straightforward clone of Sublime in many ways.

Task running - I would suggest using the gulp integration instead of running npm.  It auto-detects gulp tasks, so you don't have to list them in `tasks.json` except to add information.  Locally, I deleted and recreated the tasks file using Configure Task Runner and selecting Gulp.

I dug into the build-watch situation you describe in the document.  VSCode supports "background tasks" which seem to be for this exact purpose.  I configured the "build-watch" gulp task as `"isWatching`: true` and it avoids the spinning indicator you were seeing.

More info on background tasks:
https://code.visualstudio.com/docs/editor/tasks#_background-watching-tasks

The formatter settings aren't really right, but it seems like the configuration options are lacking as far as I can tell.

All that said, this is probably fine to go as-is, and we can improve in the future.

emackey commented:
Thanks @shunter.  I like the idea of auto-configuring gulp.

When I tried "Configure Task Runner" without a task file, it couldn't find gulp in my path and bailed out.  I searched online and some folks recommend `npm install -g gulp-cli`.  Does that sound like the right solution?  I didn't install it globally yet, as I was hoping to find a project-relative solution.

Also, if you still have that `isWatching` configuration on your machine, can you push it to this branch?  Otherwise I'll try to recreate it after getting gulp straightened out.

shunter commented:
You're right, I had gulp installed globally, which doesn't seem that unreasonable.  Seems to be the recommended instructions for working with gulp builds.  `gulp-cli` appears to just be a global bootstrapper that runs the project-local version of `gulp`.

I haven't really done any testing of these tasks beyond `build` and `build-watch` but I'll push the changes and you can see whether it works for you.

emackey commented:
Yeah I'm thinking my setup guide should recommend globally installing `gulp` or `gulp-cli`.  I guess the point of the `cli` bootstrapper is in case the project needs a particular version of gulp?  That may make it the safer global install.

Thanks for pushing the change, I'll check it out.

shunter commented:
`gulp-cli` is specifically mentioned in their docs, so I would go with that.  https://github.com/gulpjs/gulp/blob/master/docs/getting-started.md

emackey commented:
Thanks.  The latest version of VSCode warned me that `isWatching` was deprecated and replaced with `isBackground`, so I switched those and the background watcher appears to still work fine.  I updated the setup guide to reflect the new gulp install.  This is ready.

shunter commented:
:+1:
